<!DOCTYPE html>
<html lang="ja-JP">
<head>
<!-- Global Site Tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-620926-3"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments)};
  gtag('js', new Date());

  gtag('config', 'UA-620926-3');
</script>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="../../../style.css">
<link rel="stylesheet" href="../../../syntax-highlight.css">
<link rel="icon" type="image/png" href="../../../rurema.png">

<link rel="canonical" href="https://docs.ruby-lang.org/ja/latest/method/Exception/i/full_message.html">

<title>Exception#full_message (Ruby 2.6.0 リファレンスマニュアル)</title>
<meta name="description" content="返される文字列は Ruby が捕捉されなかった例外を標準エラー出力に出力するときと
同じ形式です。
そのため、メソッド呼び出し時に $stderr が変更されておらず、$stderr.tty? が真の場合は
エスケープシーケンスによる文字装飾がついています。">
<script src="../../../script.js"></script>
</head>
<body>


<script type="application/ld+json">
{"@context":"http://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","item":"https://docs.ruby-lang.org/ja/latest/doc/index.html","name":"Ruby 2.6.0 リファレンスマニュアル","position":1},{"@type":"ListItem","item":"https://docs.ruby-lang.org/ja/latest/library/index.html","name":"ライブラリ一覧","position":2},{"@type":"ListItem","item":"https://docs.ruby-lang.org/ja/latest/library/_builtin.html","name":"組み込みライブラリ","position":3},{"@type":"ListItem","item":"https://docs.ruby-lang.org/ja/latest/class/Exception.html","name":"Exceptionクラス","position":4},{"@type":"ListItem","item":"https://docs.ruby-lang.org/ja/latest/method/Exception/i/full_message.html","name":"full_message","position":5}]}
</script>

<header>
  <nav>
    <ol class="inline-breadcrumb-list">
      <li>
        <a href="../../../doc/index.html">Ruby 2.6.0 リファレンスマニュアル</a>
      </li>
      <li>
        <a href="../../../library/index.html">ライブラリ一覧</a>
      </li>
      <li>
        <a href="../../../library/_builtin.html">組み込みライブラリ</a>
      </li>
      <li>
        <a href="../../../class/Exception.html">Exceptionクラス</a>
      </li>
      <li>
        full_message 
      </li>
    </ol>
  </nav>

<h1>instance method Exception#full_message</h1>
</header>

<main>
<dl class="methodlist">

<dt class="method-heading" id="I_FULL_MESSAGE"><code>full_message(highlight: true, order: :bottom) -&gt; String</code><span class="permalink">[<a href="../../../method/Exception/i/full_message.html">permalink</a>][<a href="https://docs.ruby-lang.org/en/2.6.0/Exception.html#method-i-full_message">rdoc</a>][<a href="https://github.com/rurema/doctree/edit/master/refm/api/src/_builtin/Exception#L249">edit</a>]</span></dt>
<dd class="method-description">
<p>
例外の整形された文字列を返します。
</p>
<p>
返される文字列は Ruby が捕捉されなかった例外を標準エラー出力に出力するときと同じ形式です。そのため、メソッド呼び出し時に $stderr が変更されておらず、$stderr.tty? が真の場合はエスケープシーケンスによる文字装飾がついています。
</p>
<dl>
<dt class='method-param'>[PARAM] highlight:</dt>
<dd>
エスケープシーケンスによる文字装飾をつけるかどうかを指定します。デフォルト値は <a href="../../../method/Exception/s/to_tty=3f.html">Exception.to_tty?</a> の返り値と同じです。
</dd>
<dt class='method-param'>[PARAM] order:</dt>
<dd>
:top か :bottom で指定する必要があります。バックトレースの一番奥がエラーメッセージの上(top)か下(bottom)かを指定します。デフォルト値は <a href="../../../method/Exception/s/to_tty=3f.html">Exception.to_tty?</a> が真なら :bottom で偽なら :top です。
</dd>
</dl>
<pre class="highlight ruby">
<span class="caption">例</span>
<code>
<span class="k">begin</span>
  <span class="nb">raise</span> <span class="s2">"</span><span class="s2">test</span><span class="s2">"</span>
<span class="k">rescue</span> <span class="o">=&gt;</span> e
  <span class="nb">p</span> e<span class="p">.</span><span class="nf">full_message</span>   <span class="c1"># =&gt; &quot;\e[1mTraceback \e[m(most recent call last):\ntest.rb:2:in `&lt;main&gt;': \e[1mtest (\e[4;1mRuntimeError\e[m\e[1m)\n\e[m&quot;
</span>  <span class="vg">$stderr</span> <span class="o">=</span> <span class="vg">$stdout</span>
  <span class="nb">p</span> e<span class="p">.</span><span class="nf">full_message</span>   <span class="c1"># =&gt; &quot;test.rb:2:in `&lt;main&gt;': test (RuntimeError)\n&quot;
</span>  <span class="vg">$stderr</span> <span class="o">=</span> <span class="no">STDERR</span>
  <span class="nb">p</span> e<span class="p">.</span><span class="nf">full_message</span>   <span class="c1"># =&gt; &quot;\e[1mTraceback \e[m(most recent call last):\ntest.rb:2:in `&lt;main&gt;': \e[1mtest (\e[4;1mRuntimeError\e[m\e[1m)\n\e[m&quot;
</span><span class="k">end</span>
</code></pre>
<p>
[SEE_ALSO] <a href="../../../method/Exception/s/to_tty=3f.html">Exception.to_tty?</a>
</p>
</dd>


</dl>
</main>

<footer id="footer">
  <a rel="license" href="https://creativecommons.org/licenses/by/3.0/">
    <img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/3.0/88x31.png" />
  </a>

  <a href="https://github.com/rurema/doctree/issues/new" id="feedback-link">フィードバックを送る</a>
  
    / <a href="https://github.com/rurema/doctree/edit/master/refm/api/src/_builtin/Exception#L249">このマニュアルを編集する</a>
  
  <script>if (window.URLSearchParams) { document.getElementById("feedback-link").search = new URLSearchParams({'body': document.location}); }</script>
</footer>
</body>
</html>
